config/opal_check_pc.m4

```diff
@@ -0,0 +1,40 @@
+dnl -*- shell-script -*-
+dnl
+dnl Copyright (c) 2021 IBM Corporation.  All rights reserved.
+dnl
+dnl $COPYRIGHT$
+dnl
+dnl Additional copyrights may follow
+dnl
+dnl $HEADER$
+dnl
+
+AC_DEFUN([OPAL_CHECK_LFLAGS], [
+         AC_MSG_CHECKING([for pc file for])
+         AC_MSG_RESULT([$1])
+
+	 num_paths=$#
+         save_name=$1
+         cmd="pkg-config"
```

Comment:
This adds an explicit dependency on pkg-config for building, which is new to Open MPI and something I know Jeff was worried about when we talked about it earlier in the week.  We should consider parsing the .pc file directly (as awful as that is, we at least can vaguely contain the pain because we're generating the pc files).

Reply:
I understand not wanting to add another dependency, but trying to parse the file may not be quite good enough - or at worst will be difficult to maintain as things change.

For example, PMIx has a dependency on zlib on our machines. This would require adding `-lz` for static builds, however pmix.pc contains only `requires zlib` as a hint.

Now - for this specific example, it might be easy enough to do that translation, but I don't know if there is a repeatable/defined way for determining the correct linker flag from the library name. It sounds like a maintenance headache, but maybe there is a right way to do it?

Then again, this is exactly what pkg-config is supposed to do, and re-inventing the wheel probably isn't the right approach.

Reply:
As a smaller hammer approach, ompi could only require pkg-config for the `--enable-static` configure line.

Reply:
actually, there's no difference between --enable-static or --enable shared here (at least for the external case).  What matters is whether or not PMIX/PRRTE are built with static or not.

Reply:
I don't necessarily have objection to using pkg-config directly.  Maybe as a default path, and have an override so users could handle the edge cases or not having pkg-config.  But this is something we should talk about with more input than just you and I :).

Reply:
I am a bit concerned about requiring `pkg-config`.  It's not present in MacOS, for example.  I don't know if it's available by default in various Linux distros.

Is it really that difficult to parse a `.pc` file?  It's just key=value.  Is it more complicated than `egrep 'THE_RIGHT_KEY\s*=' | cut -f\  -d2`? (disclaimer: typed directly in github and not tested)

Reply:
I don't have a preference either.

@jsquyres we could do it, but do we have a list of all libraries key->value translation that could pop up? See my example on zlib. The pmix.pc file only says, `requires: zlib`, which would require us knowing that an `-lz` is needed for a static build:

```
$. cat pmix.pc
exec_prefix=${prefix}
libdir=${exec_prefix}/lib
includedir=${prefix}/include

Name: pmix
Description: Process Management Interface for Exascale (PMIx)
Version: 5.0.0a1
URL: https://pmix.org/
Requires:  zlib
Libs: -L${libdir} -lpmix 
Cflags: -I${includedir}
```
This is just one example...I don't know of all the translations that would need to be done for every environment. 

Now, if we know all of these, and it's as easy as forming a key -> value table, that would work. I'm not sure what the maintenance would be on that longterm...

You mentioned in the ticket that your hwloc has a dependency on -lpciaccess - that's not something that is required when I build on my machines. It makes me wonder what other things are out there in various environments/setups that will be missed.

Reply:
Well, looking at it again I suppose it could be done:

`Requires: zlib foolib` -> `-lz -lfoo` __should__ work if we don't want the pkg-config dependency.

Reply:
I don't think we want to do the requires -> libs change, because we'd lose the recursive lookup behavior.  I kind of like my idea:

1) if pkg-config exists, use it (ie, this CR with all the fixes requested)
2) if pkg-config doesn't exist, assume pmix, prrte, hwloc, libevent, etc. are dynamic linked or have no extra dependent libraries
3) offer a --with-{prrte,pmix,hwloc,libevent}-extra-libs flag so that customers who don't have pkg-config and run into problems with 2 have an exit path.

@jsquyres thoughts?

Reply:
Right, it would be almost impossible for ompi to get the required libs down the line.

That suggestion sounds fine to me. Should it just be a generic `--with-extra-libs` flag? I don't think it really matters that it gets differentiated for purposes of this PR, just as long as they are tacked onto `OMPI_DEPS_LFLAGS`